src/client/connection.ts

```diff
@@ -43,40 +44,56 @@ export function getEventRouter(): EventRouter<Events> {
 /**
  * Promise to the channel object that allows us to connect to the client
  */
-let channelPromise: Promise<ChannelClient>|null = null;
+let channelPromise: Promise<ChannelClient> | null = null;
+const hasDOMContentLoaded = new DeferredPromise<void>();
 
-if (typeof fin !== 'undefined') {
-    getServicePromise();
-}
+if (typeof document !== 'undefined') {
+    document.addEventListener('DOMContentLoaded', () => {
+        hasDOMContentLoaded.resolve();
```

Comment:
Suggest put this beneath the below block, so it's 100% clear that by the time the line `await hasDOMContentLoaded.promise`, `channelPromise` may be set